server/iframe.go

```diff
@@ -72,13 +76,41 @@ func (a *API) authenticate(w http.ResponseWriter, r *http.Request) {
 	var logger logrus.FieldLogger
 	logger = logrus.StandardLogger()
 
+	redirectPath := "/"
+
+	// Check if we have a subEntityID coming from the Microsoft Teams SDK to redirect the user to the correct URL.
+	// We use this from the Team's notifications to redirect the user to what triggered the notification, in this case,
+	// a post.
+	subEntityID := r.URL.Query().Get("sub_entity_id")
+	if subEntityID != "" {
+		if strings.HasPrefix(subEntityID, "post_") {
+			postID := strings.TrimPrefix(subEntityID, "post_")
+			post, err := a.p.API.GetPost(postID)
+			if err != nil {
+				logger.WithError(err).Error("Failed to get post to generate redirect path from subEntityId")
```

Comment:
Already commented in other PR, but nil dereference potential here unless we break out.

Reply:
Fixed this on #806 where the redirect logic was isolated into it's own function: fc423b288e6a9ff21ebd382d96577e12959d66a8